src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -248,32 +263,48 @@ public PolicyEntity call() throws Exception {
      * @return Domain
      */
     @POST
-    public DomainEntity createDomain(DomainEntity entity) {
+    public String createDomain(DomainEntity entity) {
 
         logger.info("Creating Domain Entity...:" + entity.getName());
 
-        return this.txControl.required(new Callable<DomainEntity>() {
+        return this.txControl.required(new Callable<String>() {
 
             @Override
-            public DomainEntity call() throws Exception {
+            public String call() throws Exception {
 
-                DomainEntity domainElement = new DomainEntity();
-                CriteriaBuilder cb = em.getCriteriaBuilder();
+                DomainEntity domainElement = null;
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
 
-                CriteriaQuery<ApplianceManagerConnectorEntity> q = cb
+                CriteriaQuery<ApplianceManagerConnectorEntity> query = criteriaBuilder
                         .createQuery(ApplianceManagerConnectorEntity.class);
-                Root<ApplianceManagerConnectorEntity> r = q.from(ApplianceManagerConnectorEntity.class);
-                q.select(r)
-                        .where(cb.and(cb.equal(r.get("name"), entity.getapplianceManagerConnector().getName())));
+                Root<ApplianceManagerConnectorEntity> r = query.from(ApplianceManagerConnectorEntity.class);
+                query.select(r)
+                        .where(criteriaBuilder.and(
+                                criteriaBuilder.equal(r.get("name"), entity.getapplianceManagerConnector().getName())));
+
+                List<ApplianceManagerConnectorEntity> result = em.createQuery(query).getResultList();
+
+                CriteriaQuery<DomainEntity> domainquery = criteriaBuilder.createQuery(DomainEntity.class);
+                Root<DomainEntity> rdomain = domainquery.from(DomainEntity.class);
+                domainquery.select(rdomain)
+                        .where(criteriaBuilder
+                        .and(criteriaBuilder.equal(r.get("name"), entity.getName())));
+
+                List<DomainEntity> domainresult = em.createQuery(domainquery).getResultList();
+                if (!domainresult.isEmpty()) {
+                    throw new Exception("Domain name already exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
```

Comment:
in this case it should not be 404, 404  is not found which does not make sense here, it should be 400  bad request

Reply:
ok